core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+
+    } else {
+        v = vec3(m_static->getGlobalPose().p.x, m_static->getGlobalPose().p.y, m_static->getGlobalPose().p.z);
+    }
+    return v;
+}
+
+quat RigidBody::get_rotation() const {
+    quat q;
+    if (m_dynamic) {
+        q = quat(m_dynamic->getGlobalPose().q.w, m_dynamic->getGlobalPose().q.x, m_dynamic->getGlobalPose().q.y,
+                 m_dynamic->getGlobalPose().q.z);
+    } else {
+        q = quat(m_static->getGlobalPose().q.w, m_static->getGlobalPose().q.x, m_static->getGlobalPose().q.y,
+                 m_static->getGlobalPose().q.z);
+    }
+    return q;
+}
+
+physx::PxMaterial* RigidBody::get_pxmaterial() {
+    return m_material;
+}
+
+void RigidBody::set_transform(const vec3& position, const quat& rotation) {
+    if (m_dynamic == NULL && m_static == NULL) {
+        return;
+    }
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_position(const vec3& position) {
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    physx::PxVec3 p = physx::PxVec3(p_x, p_y, p_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(p));
+    } else {
+        m_static->setGlobalPose(PxTransform(p));
+    }
+}
+void RigidBody::set_rotation(const quat& rotation) {
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
+    physx::PxQuat q = physx::PxQuat(r_w, r_x, r_y, r_z);
+    if (m_dynamic) {
+        m_dynamic->setGlobalPose(PxTransform(q));
+    } else {
+        m_static->setGlobalPose(PxTransform(q));
+    }
+}
+
+void RigidBody::set_material(physx::PxMaterial* material) {
+    m_material = material;
+}
+
+void RigidBody::set_physics_and_scene(physx::PxScene* scene, physx::PxPhysics* physics) {
+    m_scene = scene;
+    m_physics = physics;
+}
+
+void RigidBody::create_cube_rigid_dynamic(const vec3& halfSize,
+                                          const float& mass,
+                                          const vec3& position,
+                                          const quat& rotation) {
+    if (m_static) {
+        return;
+    }
+    physx::PxShape* shape = NULL;
+    float p_x = position.x;
+    float p_y = position.y;
+    float p_z = position.z;
+    float r_w = rotation.w;
+    float r_x = rotation.x;
+    float r_y = rotation.y;
+    float r_z = rotation.z;
```

Comment:
does this data need to be set to a float p_x p_y .... can position.x ,position.y .... be directly passed into the functions PxVec3 / PxQuat instead like it is with half_size.x